apps/emqx_authn/src/emqx_authn_initializer.erl

```diff
@@ -0,0 +1,141 @@
+%%--------------------------------------------------------------------
+%% Copyright (c) 2021 EMQ Technologies Co., Ltd. All Rights Reserved.
+%%
+%% Licensed under the Apache License, Version 2.0 (the "License");
+%% you may not use this file except in compliance with the License.
+%% You may obtain a copy of the License at
+%%
+%%     http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing, software
+%% distributed under the License is distributed on an "AS IS" BASIS,
+%% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+%% See the License for the specific language governing permissions and
+%% limitations under the License.
+%%--------------------------------------------------------------------
```

Comment:
I usually advocate for `handle_event_function` callback mode for gen_statem.
Reasons being:
1) With state function style it's easy to forget to handle a common event (e.g. unexpected call), and fixing that reliably basically means re-implementing handle_event_function 
2) handle_event_function lets you group clauses logically
3) handle_event_function-style of state machine is much easier to understand, since the actual state machine can fit on one screen and it doesn't require jumping through the file: https://github.com/emqx/mria/blob/main/src/mria_rlog_replica.erl#L100

Reply:
That is reasonable, I'll implement a `handle_event_function` variant.

Reply:
✅